config/urls.py

```diff
@@ -30,4 +30,5 @@
     path('snippets/new/', views.add_snippet, name='add_snippet'),
     path('snippets/<int:pk>/edit/', views.edit_snippet, name='edit_snippet'),
     path('snippets/<int:pk>/delete/', views.delete_snippet, name='delete_snippet'),
+    path('snippets/user_profile', views.profile, name='user_profile'),
 ]
```

Comment:
I think this morning you had this url as `snippets/<str:username>/user_profile` though you could choose to match the url for the logged in user's own profile. The whole thing might then be:
```
path('snippets/profile/<str:username>', views.user_profile, name='user_profile'),
```